src/material_library.h

```diff
@@ -0,0 +1,204 @@
+#ifndef PYNE_MATERIAL_LIBRARY
+#define PYNE_MATERIAL_LIBRARY
+
+#include <stdlib.h>
+#include <iostream>
+#include <unordered_map>
+#include <string>
+#include <string>
+#include <vector>
+
+#if !defined(JSON_IS_AMALGAMATION)
+#define JSON_IS_AMALGAMATION
+#endif
+
+#ifndef PYNE_IS_AMALGAMATED
+#include "material.h"
+#endif
+
+namespace pyne {
+
+typedef std::unordered_map<std::string, pyne::Material*> mat_map;
+typedef std::set<std::string> matname_set;
+typedef std::set<int> nuc_set;
+
+class MaterialLibrary {
+ protected:
+  // The actual library
+  mat_map material_library;  // material library
+  matname_set matlist;
+  nuc_set nuclist;
+
+
+ private:
+  /**
+   * \brief Add the nuclide form the material to the list of nuclides
+   * \param material from which add the nuclide to the list
+  */
+  void append_to_nuclist(pyne::Material mat);
+  /**
+   * \brief determines in that datapath exists in the hdf5 file
+   * \param[in] filename of the h5m file
+   * \param[in] the datapath of we would like to test
+   * \return true/false
+   */
+  bool hdf5_path_exists(const std::string& filename,
+                        const std::string& datapath);
+
+  /**
+   * \brief determines the length of an hdf5 data table
+   * \param[in] filename of the h5m file
+   * \param[in] the datapath of we would like to read
+   * \return the number of elements to the array
+   */
+  int get_length_of_table(const std::string& filename,
```

Comment:
The length of the table has a semantic purpose (number of materials?) - rename to `get_number_of_materials`

Reply:
This is actually a generic method that allows to get the length of any table in a `hdf5` file ? shall it not move to `utils` ? and keep its name... ?